pkg/cache/tas_flavor_snapshot.go

```diff
@@ -695,3 +720,19 @@ func (s *TASFlavorSnapshot) notFitMessage(fitCount, totalCount int32) string {
 	}
 	return fmt.Sprintf("topology %q allows to fit only %v out of %v pod(s)", s.topologyName, fitCount, totalCount)
 }
+
+// compileNodeSelector validates and compiles node selectors, returning a selector.
+// Returns labels.Everything() if no selectors are provided.
+// Returns labels.Nothing() if selectors are invalid.
+func (s *TASFlavorSnapshot) compileNodeSelector(nodeSelectors map[string]string) labels.Selector {
+	// If there are no node selectors match everything.
+	if len(nodeSelectors) == 0 {
+		return labels.Everything()
+	}
+	selector, err := labels.ValidatedSelectorFromSet(labels.Set(nodeSelectors))
+	if err != nil {
+		s.log.Error(err, "invalid node selectors provided, no nodes will be matched", "selectors", nodeSelectors)
+		return labels.Nothing()
```

Comment:
I think this code is unreachable currently, so it is probably better to return error instead (less code). If somehow this error is reached then it would be propagated up to the user in event, which is a good thing.